File: frontend/src/app/shared/model/authentication/dto/user.dto.ts
import { Role } from '../role.enum';

export class UserDto {
    public readonly id: number | null;
    public readonly firstname: string;
    public readonly lastname: string;
    public readonly appUsername: string;
    public readonly email: string;
    public readonly role: Role;
    public readonly timetablePreference: number | null;
    public readonly universityPreference: number | null;
    public readonly facultiesPreference: number[] | null;
    public readonly majorsPreference: number[] | null;

    constructor(
        id: number | null = null,
        firstname: string,
        lastname: string,
        appUsername: string,
        email: string,
        role: Role,
        timetablePreference: number | null,
        universityPreference: number | null,
        facultiesPreference: number[] | null,
        majorsPreference: number[] | null
    ) {
        this.id = id;
        this.firstname = firstname;
        this.lastname = lastname;
        this.appUsername = appUsername;
        this.email = email;
        this.role = role;
        this.timetablePreference = timetablePreference;
        this.universityPreference = universityPreference;
        this.facultiesPreference = facultiesPreference;
        this.majorsPreference = majorsPreference;
    }
}

File: backend/src/main/java/hu/szakdoga/backend/authentication/data/dto/UserDTO.java
package hu.szakdoga.backend.authentication.data.dto;

import hu.szakdoga.backend.authentication.data.model.Role;
import jakarta.persistence.Column;
import jakarta.persistence.ElementCollection;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@Builder
public class UserDTO {
    private final Long id;
    private final String firstname;
    private final String lastname;
    private final String appUsername;
    private final String email;
    private final Role role;
    private final Long timetablePreference;
    private final Long universityPreference;
    private final List<Long> facultiesPreference;
    private final List<Long> majorsPreference;
}
